File: videojuegoDeChinpokomon/src/logger/LoggerAux.java
package logger;

public final class LoggerAux {
    
    private static LoggerAux instance;

    private String level;

    private static String debugLevel;

    private static String infoLevel;

    private static String warnLevel;

    private static String errorLevel;

    public LoggerAux(String level) {
        this.level = level;
    }

    public static LoggerAux getInstance(String level){
        if (instance == null) {
            instance = new LoggerAux(level);
        }
        return instance;
    }

    public void setLevel(String level){
        this.level = level;
    }

    public String getLevel() {
        return level;
    }

    public String getDebug() {
        if(this.level != "info" && this.level != "warn" && this.level != "error"){
            return "debug: " + debugLevel + "info: " + infoLevel + "\n" +
                    "warn: " + "\n" + warnLevel + "error: " + "\n" + errorLevel;
        }else {
            return null;
        }
    }

    public String getInfo() {
        if(this.level != "warn" && this.level != "error"){
            return "info: " + infoLevel + "\n" + "warn: " + warnLevel + "\n" + "error: " + errorLevel;
        }else {
            return null;
        }
    }

    public String getWarn() {
        if(this.level != "error"){
            return "warn: " + warnLevel + "error: " + "\n" + errorLevel;
        }else {
            return null;
        }
    }

    public String getError() {
        return "error: " + errorLevel;
    }

    public static void debug(String message) {
        debugLevel = message;
    }

    public static void info(String message) {
       infoLevel = message;
    }

    public static void warn(String message) {
        warnLevel = message;
    }

    public static void error(String message) {
        errorLevel = message;
    }

}

File: ChimpokomonPython/ataque/LoggerAux.py

def singleton(cls):

    instances = dict()

    def wrap(*arg, **kwargs):
        if cls not in instances:
            instances[cls] = cls(*arg, **kwargs)

        return instances[cls]

    return wrap


@singleton
class LoggerAux():
    
    LEVEL_DEBUG = "DEBUG"
    LEVEL_INFO = "INFO"
    LEVEL_WARN = "WARN"
    LEVEL_ERROR = "ERROR"

    def __init__(self):
        self._level = self.LEVEL_DEBUG
        self._debug = ""
        self._info = ""
        self._warn = ""
        self._error = ""
    
    def setLevel(self, level):
        self._level = level

    
    def debug(self):
        if self._level == self.LEVEL_DEBUG:
            return "debug: " + self._debug + "info: " + self._info + "warn: " + self._warn + "error: " + self._error
        else:
            return None
        

    def info(self):
        if self._level != self.LEVEL_WARN and self._level != self.LEVEL_ERROR:
            return "info: " + self._info + "warn: " + self._warn + "error: " + self._error
        else:
            return None

    def warn(self):
        if self._level != self.LEVEL_ERROR:
            return "warn: " + self._warn + "error: " + self._error
        else:
            return None

    def error(self):
        return "error: " + self._error

    def setDebug(self, debug):
        self._debug = debug

    def setInfo(self, info):
        self._info = info

    def setWarn(self, warn):
        self._warn = warn

    def setError(self, error):
        self._error = error

LOGGER = LoggerAux()